kv/kvserver: (unknown) failed

kv/kvserver.(unknown) [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3374094&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3374094&tab=artifacts#/) on master @ [45ffbd022fb18124c610a2926dd7028a707f3f45](https://github.com/cockroachdb/cockroach/commits/45ffbd022fb18124c610a2926dd7028a707f3f45):


```
Slow failing tests:
TestLearnerSnapshotFailsRollback/non-voter - 0.00s

Slow passing tests:
TestMergeQueueSeesNonVoters - 222.76s
TestStoreRangeMergeTimestampCache - 68.65s
TestBumpSideTransportClosed - 58.97s
TestClosedTimestampFrozenAfterSubsumption - 57.33s
TestRemovePlaceholderRace - 43.97s
TestStoreLeaseTransferTimestampCacheRead - 42.88s
TestReplicaTombstone - 33.12s
TestProcessSplitAfterRightHandSideHasBeenRemoved - 32.58s
TestSplitBurstWithSlowFollower - 28.15s
TestDiscoverIntentAcrossLeaseTransferAwayAndBack - 26.81s
TestStoreRangeUpReplicate - 26.72s
TestClosedTimestampCanServeWithConflictingIntent - 26.40s
TestCheckConsistencyInconsistent - 25.82s
TestGossipHandlesReplacedNode - 25.38s
TestRangefeedWorksOnSystemRangesUnconditionally - 25.24s
TestLeaseInfoRequest - 25.01s
TestCheckConsistencyMultiStore - 24.00s
TestStoreMaxBehindNanosOnlyTracksEpochBasedLeases - 23.74s
TestSplitSnapshotRace_SnapshotWins - 22.25s
TestAdminRelocateRange - 21.42s
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=(unknown) PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*\(unknown\).*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

```
index 1059 with error: nope
    soon.go:71: encountered retriable replication change error: (n2,s2):176NON_VOTER: remote couldn't accept INITIAL snapshot 8469cd52 at applied index 1066 with error: nope
panic: test timed out after 40m0s
```

```
goroutine 916852 [sleep]:
time.Sleep(0x3b9aca00)
	/usr/local/go/src/runtime/time.go:193 +0x12e
github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration(0x0, 0xc0091e4c70)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:214 +0x85
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError(0xc0031d0d00, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:77 +0x85
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoonError(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49
github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).changeReplicas(0xc009689400, 0x2, {0xc007293b58, 0x1, 0x8}, {0xc0031d13f0, 0x1, 0x1})
	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:616 +0x1d6
github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).addReplica(0x0, {0xc007293b58, 0x1, 0x8}, 0x0, {0xc0091e53f0, 0x1, 0x1})
	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:655 +0x108
github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).AddNonVoters(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:680
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestLearnerSnapshotFailsRollback.func1(0xc0006fc4e0, 0x5)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_learner_test.go:338 +0x40e
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestLearnerSnapshotFailsRollback.func3(0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_learner_test.go:356 +0x42
testing.tRunner(0xc0006fc4e0, 0xc00762c5b0)
	/usr/local/go/src/testing/testing.go:1259 +0x230
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1306 +0x727
```

the test wants to see the "couldn't accept INITIAL snapshot" error bubble up, but the replication change seems to never return.

https://github.com/cockroachdb/cockroach/blob/f7b66bf5551633da64c74dd50543f4965a02624e/pkg/kv/kvserver/replica_learner_test.go#L337-L345

--- a/pkg/cmd/teamcity-trigger/main.go
+++ b/pkg/cmd/teamcity-trigger/main.go
@@ -106,11 +106,11 @@ func runTC(queueBuild func(string, map[string]string)) {
 			} else {
 				opts["env.COCKROACH_KVNEMESIS_STEPS"] = "10000"
 			}
-		case baseImportPath + "sql/logictest":
-			// Stress logic tests with reduced parallelism (to avoid overloading the
+		case baseImportPath + "sql/logictest", baseImportPath + "kv/kvserver":
+			// Stress heavy with reduced parallelism (to avoid overloading the
 			// machine, see https://github.com/cockroachdb/cockroach/pull/10966).
 			parallelism /= 2
-			// Increase logic test timeout.
+			// Increase test timeout to compensate.
 			testTimeout = 2 * time.Hour
 			maxTime = 3 * time.Hour
 		}

